corehq/apps/auditcare/tests/test_models.py

```diff
@@ -120,3 +150,22 @@ def save(self):
     real_save = cls.save
     with patch.object(cls, "save", save):
         yield config
+
+
+@contextmanager
+def patch_trace_id_header():
+    def assert_not_installed():
+        assert AccessAudit.trace_id_header is None, AccessAudit.trace_id_header
+        assert django_header not in mod.STANDARD_HEADER_KEYS, \
+            (django_header, mod.STANDARD_HEADER_KEYS)
```

Comment:
I'd prefer that this decorator handle the case where a `trace_id_header` is already set, rather than immediately failing -- I'm sure there are cases where a test environment would want to set that value outside of these tests.

Reply:
`assert_not_installed` is used to verify test pre- and post-conditions. Pre-conditions are asserted to confirm that the tests will not produce a false positive. Post-conditions are asserted to verify that cleanup was successful.

I suppose the pre-conditions could be made a bit more flexible with
```py
assert AccessAudit.trace_id_header != settings.AUDIT_TRACE_ID_HEADER, ...
```
I could also set `TRACE_HEADER` to a value that is less likely to collide with values that someone might use in localsettings. Would that be better?

Reply:
Yep!

Reply:
https://github.com/dimagi/commcare-hq/pull/29457/commits/1e5e206a66bd397f8605d902628eec95bf031c0b